### ISSUE TITLE
InjectDispatcher shouldn't fire on secondary constructor usage of dispatcher

### ISSUE THREAD

**Issue by josephlbarnett** [detekt/detekt#5225]
## Expected Behavior

```
class MyRepository(val dispatcher: CoroutineDispatcher) {
    @Inject // mark this constructor as the one used by guice at runtime, tests can still inject a TestDispatcher using the primary constructor
    constructor() : this(Dispatchers.IO)
}
```
is equivalent to 
```
class MyRepository(dispatcher: CoroutineDispatcher = Dispatchers.IO)
```
so should not trigger the `InjectDispatcher` rule to fire
## Observed Behavior

the above code fails detekt with `InjectDispatcher` 
## Steps to Reproduce

see above code
## Context

We want to mark the secondary constructor for Guice to use at runtime to construct the Repository with the IO dispatcher.  We do still want tests to be able to use a TestDispatcher etc.
## Your Environment

* Version of detekt used: 1.21.0
* Version of Gradle used (if applicable): n/a
* Gradle scan link (add `--scan` option when running the gradle task): n/a
* Operating System and version: ubuntu 22.04
* Link to your project (if it's a public repository):

**Comment by cortinico** [comment]
Yup this looks like a false-positive to me 👍 It's up for grab
